mlir/include/mlir/Dialect/Vector/IR/VectorOps.h

```diff
@@ -160,6 +160,12 @@ getAsConstantIndexOps(ArrayRef<Value> values);
 // Vector Masking Utilities
 //===----------------------------------------------------------------------===//
 
+/// Infers the mask type for a transfer op given its vector type and
+/// permutation map. The mask in a transfer op operation applies to the
+/// tensor/buffer part of it and its type should match the vector shape
+/// *before* any permutation or broadcasting.
```

Comment:
Now that this becomes part of the 'public' interface, I think it would be nice to have an example